CommandScreen.c

```diff
@@ -0,0 +1,74 @@
+#include "CommandScreen.h"
+
+#include "config.h"
+#include "CRT.h"
+#include "IncSet.h"
+#include "ListItem.h"
+#include "Platform.h"
+#include "StringUtils.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+
+static void CommandScreen_addLine(InfoScreen* this, char* line, const char* p, int line_offset, int len) {
+   memcpy(line, p - line_offset, len);
+   line[len] = '\0';
+   InfoScreen_addLine(this, line);
+}
+
+static void CommandScreen_scan(InfoScreen* this) {
+   Panel* panel = this->display;
+   int idx = MAXIMUM(Panel_getSelectedIndex(panel), 0);
+
+   Panel_prune(panel);
+
+   const char* p = this->process->comm;
+   char* line = xMalloc(COLS + 1);
+   int line_offset = 0, last_spc = -1, len;
+   for (; *p != '\0'; p++, line_offset++) {
+      if (*p == ' ') last_spc = line_offset;
+
+      if (line_offset == COLS) {
+         len = (last_spc == -1) ? line_offset : last_spc;
+         CommandScreen_addLine(this, line, p, line_offset, len);
+         line_offset -= len;
+         last_spc = -1;
+      }
+   }
```

Comment:
My head hurts trying to read this … I'd prefer if `CommandScreen_addLine` didn't input the end of the string it should add, but its beginning. With some clever restructuring this even removes the need for `CommandScreen_addLine`:

```c
   char* line = xCalloc(COLS + 1, sizeof(char));
   int line_offset = 0
   int last_space = -1;
   for (const char* p = this->process->comm; *p != 0; p++) {
      if (*p == ' ') {
         last_spc = line_offset;
      }

      line[line_offset++] = *p;

      if (COLS == line_offset) {
         if(-1 != last_space) {
            line[last_space] = 0;
            line_offset = last_space;
         }
         InfoScreen_addLine(this, line);
         if(COLS != line_offset) {
            memmov(line, line + line_offset, COLS - line_offset);
         }
         line_offset = COLS - line_offset;
         last_space = -1;
      }
   }
   if(0 != line_offset) {
      line[line_offset] = 0;
      InfoScreen_addLine(this, line);
   }
   free(line);
```
(Implementation untested, but should do)

Reply:
Hey @BenBE, thanks! And yeah, it's definitely an option to [fill the line while
scanning][1]. Just make sure to correctly [carrier over][2] the already scanned
character(s) to the next line.

On the other hand by doing that, the possible optimization in **memcpy**, namely
filling the line by words no longer applies. But I'm not sure if that's really 
a gain or lose before further checking the detail, or that doesn't even matter?

BTW, in your comment you mentioned the idea to skip leading whitespace(s) when
starting next line, I'd prefer not to do that, because spaces in a quoted string
ought to be preserved. And by not doing that we can be sure that the original
command can be reconstructed by just removing the line endings. Anyway that's not 
really a big deal.


[1]: https://github.com/ryenus/htop/commit/7f955db52fb602303711714df0a3387d8fc0e941#diff-5e605b3d4fdff9c400a8330b918626feR20-R43
[2]: https://github.com/ryenus/htop/commit/7f955db52fb602303711714df0a3387d8fc0e941#diff-5e605b3d4fdff9c400a8330b918626feR34

Reply:
Hey @BenBE, thanks! And yeah, it's definitely an option to [fill the line while
scanning][1]. Just make sure to correctly [carrier over][2] the already scanned
character(s) to the next line.

On the other hand by doing that, the possible optimization in **memcpy**, at 
least filling the line by words no longer applies. But I'm not sure if that's 
really a gain or lose before further checking the detail, or that doesn't even 
matter?

BTW, in your comment you mentioned the idea to skip leading whitespace(s) when
starting next line, I'd prefer not to do that, because spaces in a quoted string
ought to be preserved. And by not doing that we can be sure that the original
command can be reconstructed by just removing the line endings. Anyway that's not 
really a big deal.


[1]: https://github.com/ryenus/htop/commit/7f955db52fb602303711714df0a3387d8fc0e941#diff-5e605b3d4fdff9c400a8330b918626feR20-R43
[2]: https://github.com/ryenus/htop/commit/7f955db52fb602303711714df0a3387d8fc0e941#diff-5e605b3d4fdff9c400a8330b918626feR34

Reply:
OK, I [gave it a try][1] and measured the time, for a command with 7k+ 
characters, it's about `30` vs `28` (ms) on average before and after in
the best case. And the copy-on-scan approach won with a little margin.

[1]: https://github.com/ryenus/htop/commit/d96a931de437ea027d51af393523b21d4b2bea25